### PR TITLE
Added showing connection errors when using self-hosted server

### PR DESCRIPTION
**Pull request by konrad-g**
When I tired to connect from the Android client to my self-hosted server, screen indefinitely froze on "Connecting...". When I tired debugging the issue, I found out that the app only shows connection issues when host is not found, and otherwise tries silently reconnect. I understand that it might make sense in case of using managed "supla.org" servers. Still, when using self-hosted solution, it leaves you scratching your head and wondering what is going on. That is why, I added showing connection errors when host doesn't belong to "supla.org".

Example connection error screens:

cant-connect-to-host

bad-credentials

### CODE REVIEW

File: app/src/main/java/org/supla/android/lib/SuplaClient.java [hunk 1]
```diff
@@ -764,7 +771,9 @@ private void onConnError(SuplaConnError connError) {
         msg.setConnError(connError);
         sendMessage(msg);
 
-        if (connError.Code == SuplaConst.SUPLA_RESULTCODE_HOSTNOTFOUND) {
+        String host = this.getHost();
+
+        if (connError.Code == SuplaConst.SUPLA_RESULTCODE_HOSTNOTFOUND || !host.contains(".supla.org")) {
```
Comment (przemyslawzygmunt) [inline app/src/main/java/org/supla/android/lib/SuplaClient.java:776]: It is not a good idea to cancel a thread for all version-related bugs. There are errors that require reconnection.

**Review by przemyslawzygmunt — CHANGES_REQUESTED** [review]

**Comment by przemyslawzygmunt** [comment]
We only accept PRs for branch develop.

File: app/src/main/java/org/supla/android/lib/SuplaClient.java (hunk 1 above)
Comment (konrad-g) [reply app/src/main/java/org/supla/android/lib/SuplaClient.java:776]: I get it, but it's also the core of the problem. As a user I got to experience a few times getting stuck indefinitely on the "Connecting..." screen. Potentially we could add an error counter and once a limit of errors is reached (e.g. 10 errors per connection attempt), we show the error screen to the user. In that case we could also probably remove my host name check. What do you think?

Comment (przemyslawzygmunt) [reply app/src/main/java/org/supla/android/lib/SuplaClient.java:776]: This PR solves one problem and creates another. What was the real reason the application was unable to connect to your server?

**Comment by konrad-g** [comment]
I would happily contribute under the repository license ([Link to Github T&S](https://docs.github.com/en/github/site-policy/github-terms-of-service#6-contributions-under-repository-license)), getting lawyers involved makes my a bit anxious.

File: app/src/main/java/org/supla/android/lib/SuplaClient.java (hunk 1 above)
Comment (konrad-g) [reply app/src/main/java/org/supla/android/lib/SuplaClient.java:776]: Closed port 2015. It threw error: Can't connect to host

**Pull request closed without merging**